sklearn/model_selection/_split.py

```diff
@@ -2460,6 +2461,23 @@ def check_cv(cv=5, y=None, *, classifier=False):
     return cv  # New style cv objects are passed without any modification
 
 
+@validate_params(
+    {
+        "test_size": [
+            Interval(numbers.Real, 0, 1, closed="neither"),
+            Interval(numbers.Integral, 0, None, closed="neither"),
```

Comment:
Nitpick but when it's an interval of int I find more natural to include the bound when it's a closed bound
```suggestion
            Interval(numbers.Integral, 1, None, closed="left"),
```